libs/core/langchain_core/messages/tool.py

```diff
@@ -152,6 +157,19 @@ class ToolCallChunk(TypedDict):
     """An identifier associated with the tool call."""
     index: Optional[int]
     """The index of the tool call in a sequence."""
+    type: NotRequired[Literal["tool_call_chunk"]]
+
+
+def tool_call_chunk(
```

Comment:
why is this needed?

Reply:
convenience / to make sure ppl don't mess up the type

Reply:
How is that easier than using ToolCallChunk? Is it validating that there are no extra arguments or typos in the arguments?

Reply:
type is hard coded